src/pages/ProductList/ProductList.js

```diff
@@ -1,13 +1,167 @@
-import React, { Component } from 'react'
+import React, { Component } from "react";
+import Product from "./Components/Product"
+import "./ProductList.scss"
 
 class ProductList extends Component {
-    render() {
-        return (
-            <div>
-                
-            </div>
-        )
+  constructor() {
+    super();
+    this.state = {
+      filterList: [],
+      allProducts: [],
+      productsByCategory: [],
+      productsByPage: [],
+      mappingPage: false,
+      isBtnVisible: true,
+      isPageFooterVisible: true,
+      offset: 0
     }
-}
+  }
 
-export default ProductList
+  componentDidMount() {
+    const LIMIT = 12;
+    const APIOfProductFilterList = `http://localhost:3000/Data/productFilterList.json`;
+    const APIOfProductList = `http://localhost:3000/Data/productList.json?limit=${LIMIT}&offset=${this.state.offset}`;
+    // pagination(limit, offset)은 추후 시도 예정입니다.
+
+    Promise.all([
+      fetch(APIOfProductFilterList, {method: "GET"})
+        .then(res => res.json())
+        .then(res => {
+          this.setState({
+            filterList: res.filterList,
+          });
+        }).catch(error => console.log(error.message)),
+
+      fetch(APIOfProductList, {method: "GET"})
+        .then(res => res.json())
+        .then(res => {
+          this.setState({
+            allProducts: res.products,
+            productsByCategory: res.products
+          })
+        })
+    ])
+  }
+
+  filterByCategory = (category) => {
+    if (category === "ALL") {
+      this.setState({
+        productsByCategory: [...this.state.allProducts],
+        mappingPage : false,
+        isBtnVisible: true,
+        isPageFooterVisible: true
+      })
+    } 
+    else {
+      const filtering = this.state.allProducts.filter(product => {
+        return product.category === category;
+      })
+      if (filtering.length < 13) {
+        this.setState({isPageFooterVisible: false})
+      } else {
+        this.setState({isPageFooterVisible: true})
+      }
+      this.setState({
+        productsByCategory: [...filtering],
+        productsByPage: [...filtering]
+      })
+    }
+  }
+
+  filterByPage = (num) => {
+    const { productsByCategory } = this.state;
+    if (num === 1) {
+      this.setState({ 
+        productsByPage : productsByCategory.slice(0, 12),
+        mappingPage : true 
+      })
+    }
+    if (num === 2) {
+      this.setState({ 
+        productsByPage : productsByCategory.slice(12),
+        mappingPage : true })
+    }
+  }
```

Comment:
```suggestion
  filterByPage = (num) => {
    const { productsByCategory } = this.state;
    const slicePageIdx = num === 1 ? [0, 12] : [12]
    this.setState({ 
      productsByPage : productsByCategory.slice(...slicePageIdx),
      mappingPage : true 
    })
  }
```
함수 argument는 유사 배열 형태로 활용할 수도 있습니다

Reply:
와 리뷰 주신 부분 이해하려고 검색해보다가, '유사 배열 객체'라는 걸 처음으로 알게 됐어요
지금은 3, 4 페이지도 생긴 상태라, 아래와 같이 수정했습니다! 
감사합니다 :)

~~~
const { productsByCategory } = this.state;
    const slicePageIdx = num === 1 ? [0, 12] : num === 2 ? [12, 24] : num === 3 ? [24, 36] : [36];
    const prevBtnBool = num === 1 ? false : num === 2 || num === 3 ? true : true;
    const nextBtnBool = num === 1 ? true : num === 2 || num === 3 ? true : false;

    this.setState({
      productsByPage : productsByCategory.slice(...slicePageIdx),
      mappingPage : true,
      isPrevBtnVisible : prevBtnBool,
      isNextBtnVisible : nextBtnBool
    })
~~~

Reply:
@JeanBaek 충분히 좋은 코드인 것 같습니다. 다만 페이지 수가 증가함에 따라 앞서 선언해 둔 세 변수의 조건식이 복잡해지는 문제가 생기겠네요. 이 부분은 보다 나은 형태로 바꿀 수 있도록 고민해보세요. 고생하셨습니다:)

Reply:
넵! 2차 스프린트에서 계속 고민해보겠습니다 :)
감사합니다!!